website/server/controllers/api-v3/tasks/groups.js

```diff
@@ -122,7 +122,7 @@ api.assignTask = {
     let group = await Group.getGroup({user, groupId: task.group.id, fields: requiredGroupFields});
     if (!group) throw new NotFound(res.t('groupNotFound'));
 
-    if (group.leader !== user._id) throw new NotAuthorized(res.t('onlyGroupLeaderCanEditTasks'));
+    if (group.leader !== user._id && user._id !== assignedUserId) throw new NotAuthorized(res.t('onlyGroupLeaderCanEditTasks'));
```

Comment:
This means that the user has to be the leader or they have to be assigning themselves. 

assignedUserId is the user that is being assigned. So, if the user._id is equal to the user being assigned, they are assigning themselves. 

You can also look at the opposite logic: (!condition1 && !condition2) = (condition1 || condition2) if that helps...